src/resonate/retry_policy.py

```diff
@@ -0,0 +1,102 @@
+from __future__ import annotations
+
+from abc import ABC, abstractmethod
+from dataclasses import dataclass
+from typing import Union, final
+
+from typing_extensions import TypeAlias
+
+
+class Retriable(ABC):
+    @abstractmethod
+    def calculate_delay(self, attempt: int) -> float: ...
+
+    @abstractmethod
+    def should_retry(self, attempt: int) -> bool: ...
+
+    @abstractmethod
+    def total_possible_delay(self) -> float: ...
+
+
+@final
+@dataclass(frozen=True)
+class Exponential(Retriable):
+    """A retry policy where the delay between retries grows exponentially."""
+
+    base_delay: float
+    factor: float
+    max_retries: int
+
+    def calculate_delay(self, attempt: int) -> float:
+        return self.base_delay * (self.factor**attempt)
+
+    def should_retry(self, attempt: int) -> bool:
+        return attempt <= self.max_retries
+
+    def total_possible_delay(self) -> float:
```

Comment:
Since this method is not used for prod. Let's define a helper function in the test file, so we don't create any doubts